src/mono/mono.proj

```diff
@@ -60,6 +60,11 @@
     <Error Condition="'$(TargetsAndroid)' == 'true' and '$(ANDROID_NDK_ROOT)' == ''" Text="Error: You need to set the ANDROID_NDK_ROOT environment variable pointing to the Android NDK root." />
   </Target>
 
+  <!-- CI specific build options -->    
+  <ItemGroup Condition="'$(ContinuousIntegrationBuild)' == 'true' and ('$(TargetsOSX)' == 'true' or '$(TargetsBrowser)' == 'true')">
```

Comment:
I'm not a huge fan of making the CI build do something without giving an easy way to do the same thing in a local build.

tl;dr - we should have a property that you can pass to `build.sh` to treat warnings as errors. (Maybe there's already a way to pass `-DENABLE_WERROR=1` - I'm not sure, I didn't check)

Reply:
You should see the exact warnings in a local build. I'm not changing the warnings we use, just whether they're going to make the build fail or not, so I don't think it's a big deal. I'm not sure how easy it is to thread `-DENABLE_WERROR` down from `build.sh`, though it's easy when you're invoking CMake directly.

Reply:
> I'm not sure how easy it is to thread `-DENABLE_WERROR` down from `build.sh`, though it's easy when you're invoking CMake directly.

is the workflow of invoking cmake directly documented somewhere?  IMO we should make it as easy as possible for devs to fix warnings-as-errors problems locally.

Reply:
Not sure, but again you can just fix the warnings; the issues reported are the same, it's just a question of whether they block the build. I don't think it's worth holding this PR up for that and we should open a separate issue.

Anyway, there was substantial resistance to making -Werror the default locally but I'd be happy to see a build.sh flag for it. I think CoreCLR has -Werror on CI, and I'm not sure how they handle this, but maybe such an addition would be mutually beneficial.

Reply:
We can add an option to mono.proj if needed to pass more flags to cmake, but it's a separate issue.